python/kvikio/tests/test_benchmarks.py

```diff
@@ -78,3 +80,60 @@ def test_zarr_io(run_cmd, tmp_path, api):
         cwd=benchmarks_path,
     )
     assert retcode == 0
+
+
+def skipif_libcudf_s3_io_option_is_not_available() -> None:
+    """Call pytest.skip() if cudf or its "libcudf_s3_io" option isn't available
+
+    See <https://github.com/rapidsai/cudf/pull/16499>
+    """
+    cudf = pytest.importorskip("cudf")
+    try:
+        cudf.get_option("libcudf_s3_io")
+    except KeyError:
+        pytest.skip(
+            """cudf doesn't has the "libcudf_s3_io" option, """
+            "see <https://github.com/rapidsai/cudf/pull/16499>"
+        )
+
+
+@pytest.mark.parametrize(
+    "api",
+    [
+        "cupy-kvikio",
+        "numpy-kvikio",
+        "cudf-kvikio",
+        "cudf-fsspec",
+    ],
+)
+def test_aws_s3_io(run_cmd, api):
+    """Test benchmarks/aws_s3_io.py"""
+
+    if not kvikio.is_remote_file_available():
+        pytest.skip(
+            "cannot test remote IO, please build KvikIO with with AWS S3 support"
+        )
+    # Fail early if benchmark dependencies isn't available
+    import boto3  # noqa: F401
+    import moto  # noqa: F401
+
+    if "cudf" in api:
+        skipif_libcudf_s3_io_option_is_not_available()
+
+    retcode = run_cmd(
+        cmd=[
+            sys.executable or "python",
```

Comment:
I think `sys.executable` should always be defined? Are there cases where that isn't safe?